### PR TITLE
client/asset/{btc,dcr}: DCR native wallet TxHistory

### PR DESCRIPTION
**Pull request by martonp**
This diff implements TxHistory for the DCR native wallet. The BadgerTxDB used for BTC is exported and reused in DCR.

### CODE REVIEW

**Comment by martonp** [comment]
Changing this to draft until #2666 is merged.

File: client/asset/dcr/dcr.go
Comment (martonp) [reply client/asset/dcr/dcr.go]: Removed.

File: client/asset/dcr/dcr.go
Comment (buck54321) [inline client/asset/dcr/dcr.go:5568]: This isn't useful.

File: client/asset/dcr/dcr.go
Comment (buck54321) [inline client/asset/dcr/dcr.go:5598]: Can delete?

File: client/asset/dcr/dcr.go
Comment (buck54321) [inline client/asset/dcr/dcr.go:5647]: Why not just do everything from here down under lock?

File: client/asset/interface.go
Comment (buck54321) [inline client/asset/interface.go:1090]: TicketRevo**c**ation

**Review by buck54321 — CHANGES_REQUESTED** [review]

File: client/asset/dcr/dcr.go
Comment (martonp) [reply client/asset/dcr/dcr.go:5598]: Sorry about the logs.

File: client/asset/dcr/dcr.go
Comment (martonp) [reply client/asset/dcr/dcr.go:5647]: I didn't want it to hold the `pendingTxsMtx` for too long. It would hold up every other function that submits a transaction.

**Review by buck54321 — APPROVED** [review]

**Pull request merged**
